### DISCUSSION TITLE
How can I set the L2 Cache's lower translate to DTLB instead of STLB?

### DISCUSSION THREAD

**Discussion by icuuuuuuer** [ChampSim/ChampSim#558] [Q&A]
I noticed that when I implement a prefetcher in the L2 Cache and set it to virtual prefetching, the translated prefetch addresses seem to be sent to the STLB by the L2 Cache. This isn't what I want because the STLB's translation cycle is too long. My ideal solution would be to add a small TLB next to the L2 Cache, but that might be too complex. Is there a way to make the virtual prefetch from the L2 Cache go through the DTLB instead of the STLB? Would I need to modify the Python scripts in the ‘/config’ directory to achieve this?

**Comment by ngober** [comment]
The relevant section of the documentation you're looking for is [this one](https://champsim.github.io/ChampSim/master/Creating-a-configuration-file.html#heterogeneous-systems). Look especially at the fourth example. Your configuration will look something like:

```
{
  "L2C": { "lower_translate": "my_tlb" },
  "caches" : [
    { "name": "my_tlb" }
  ]
}
```

You may need to work on it a bit in order to get the hierarchy to work together, like specifying the lower level for `"my_tlb"`...

```
{
  "L1D": { "lower_translate": "my_dtlb" },
  "L1I": { "lower_translate": "my_itlb" },
  "L2C": { "lower_translate": "my_tlb" },
  "caches" : [
    { "name": "my_dtlb", "lower_level": "second_level_tlb" },
    { "name": "my_itlb", "lower_level": "second_level_tlb" },
    { "name": "my_tlb", "lower_level": "second_level_tlb" },
    { "name": "second_level_tlb" }
  ]
}
```

Something along these lines should work.
